cms/static/cms/js/modules/cms.navigation.js

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright https://github.com/divio/django-cms
+ */
+
+// #############################################################################
+// NAMESPACES
+/**
+ * @module CMS
+ */
+var CMS = window.CMS || {};
+
+// #############################################################################
+// Toolbar
+(function ($) {
+    'use strict';
+
+    // shorthand for jQuery(document).ready();
+    $(function () {
+        /**
+         * Responsible for creating usable navigation for narrow screens.
+         *
+         * @class Navigation
+         * @namespace CMS
+         * @uses CMS.API.Helpers
+         */
+        CMS.Navigation = new CMS.Class({
+
+            implement: [CMS.API.Helpers],
+
+            initialize: function initialize() {
+                this._setupUI();
+                this.getWidths();
+
+                /** The zero based index of the right-most visible menu item of the left toolbar part. */
+                this.rightMostItemIndex = this.items.left.length - 1;
+
+                /** The zero based index of the left-most visible item of the right toolbar part. */
+                this.leftMostItemIndex = 0;
+
+                this._events();
+            },
+
+            /**
+             * Cache UI jquery objects
+             * @method _setupUI
+             * @private
+             */
+            _setupUI: function _setupUI() {
+                var trigger = $('.cms-toolbar-more');
+                this.ui = {
+                    window: $(window),
+                    toolbarLeftPart: $('.cms-toolbar-left'),
+                    toolbarRightPart: $('.cms-toolbar-right'),
+                    trigger: trigger,
+                    dropdown: trigger.find('> ul'),
+                    toolbarTrigger: $('.cms-toolbar-trigger'),
+                    logo: $('.cms-toolbar-item-logo')
+                };
+            },
+
+            /**
+             * Setup resize handler to construct the dropdown
+             * @method _events
+             * @private
+             */
+            _events: function _events() {
+                this.ui.window.on('resize.cms load.cms', CMS.API.Helpers.throttle(
+                    this._handleResize.bind(this), 50
+                ));
+            },
+
+            /**
+             * Calculates all the movable menu items widths
+             * @method _getWidths
+             */
+            getWidths: function getWidths() {
```

Comment:
don't see why not. if you change the widths of items (idk, change text) you might want to call this again